crates/cli-support/src/js/mod.rs

```diff
@@ -118,6 +118,58 @@ const INITIAL_HEAP_VALUES: &[&str] = &["undefined", "null", "true", "false"];
 // Must be kept in sync with `src/lib.rs` of the `wasm-bindgen` crate
 const INITIAL_HEAP_OFFSET: usize = 32;
 
+lazy_static! {
+    static ref JS_KEYWORDS: HashSet<&'static str> = {
+        let mut keywords = HashSet::new();
+        keywords.insert("await");
+        keywords.insert("break");
+        keywords.insert("case");
+        keywords.insert("catch");
+        keywords.insert("class");
+        keywords.insert("const");
+        keywords.insert("continue");
+        keywords.insert("debugger");
+        keywords.insert("default");
+        keywords.insert("delete");
+        keywords.insert("do");
+        keywords.insert("else");
+        keywords.insert("enum");
+        keywords.insert("export");
+        keywords.insert("extends");
+        keywords.insert("false");
+        keywords.insert("finally");
+        keywords.insert("for");
+        keywords.insert("function");
+        keywords.insert("if");
+        keywords.insert("implements");
+        keywords.insert("import");
+        keywords.insert("in");
+        keywords.insert("instanceof");
+        keywords.insert("interface");
+        keywords.insert("new");
+        keywords.insert("null");
+        keywords.insert("package");
+        keywords.insert("private");
+        keywords.insert("protected");
+        keywords.insert("public");
+        keywords.insert("return");
+        keywords.insert("super");
+        keywords.insert("switch");
+        keywords.insert("this");
+        keywords.insert("throw");
+        keywords.insert("true");
+        keywords.insert("try");
+        keywords.insert("typeof");
+        keywords.insert("undefined");
```

Comment:
As weird as it may sound, `undefined` isn't a reserved word.

Reply:
:) Yeah, I know it's a global property, but I thought it's something we would like to leave here. It's totally legal to have such property name, so in the very rare case it can prevent breaking code.